admin/class-yoast-form.php

```diff
@@ -153,7 +153,28 @@ public function set_options_value( $key, $value, $overwrite = false ) {
 	 */
 	public function admin_footer( $submit = true, $show_sidebar = true ) {
 		if ( $submit ) {
+			$floating_button = new WPSEO_Admin_Saved_Changes_Listener();
+			echo '<div id="wpseo-submit-container">';
+
+			echo '<div id="wpseo-submit-container-float" class="wpseo-admin-submit">';
+			submit_button( __( 'Save changes', 'wordpress-seo' ) );
+			if ( $floating_button->have_settings_been_saved() ) {
+				echo '<p class="message"><span class="dashicons dashicons-yes"></span>',
+				esc_html__( 'Settings saved.', 'wordpress-seo' ),
+				'</p>';
+			}
+			echo '</div>';
+
+			echo '<div id="wpseo-submit-container-fixed" class="wpseo-admin-submit wpseo-admin-submit-fixed">';
 			submit_button( __( 'Save changes', 'wordpress-seo' ) );
+			if ( $floating_button->have_settings_been_saved() ) {
```

Comment:
Have you considered moving the logic in this else to its own method in the `WPSEO_Admin_Saved_Changes_Listener` class? For example:
```php
public function show_success_state() {
    if ( $this->have_settings_been_saved() ) {
       echo '<p class="message"><span class="dashicons dashicons-yes"></span>',
					esc_html__( 'Settings saved.', 'wordpress-seo' ),
					'</p>';
    }
```

Reply:
By having such method you just tell and don't ask (the success state in this case)